### DISCUSSION TITLE
How to mesh with SimModeler  using .stl file

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#910] [Q&A] [answered]
The details are [here](https://github.com/SeisSol/SeisSol/discussions/878#:~:text=I%20noticed%20that,puml.h5%20format.)).
I've noticed that it seems that Pumgen can call SimModeler's libraries directly to generate meshes directly, which avoids the need to use gmsh to generate intermediate meshes for .msh, and allows for the direct generation of target meshes for puml.h5.

The [example](https://seissol.readthedocs.io/en/latest/meshing-with-pumgen.html#:~:text=pumgen%20%2Ds%20simmodsuite%20%2Dl%20SimModelerLib.lic%20%2D%2Dmesh%20%22Mesh%20case%201%22%20%2D%2Danalysis%20%22Analysis%20case%201%22%20test.smd%20test) you provide doesn't use a stl format file, and is there any example use stl file to generate 3d meshes?

**Comment by sebwolf-de** [comment, marked as answer]
PUMGen does not directly work on STL files, but if using the SimModeler workflow, it reads an `smd` file.
The workflow ist as follows: 
* Obtain a license and download SimModeler from www.simmetrix.com/
* Follow our documentation to create an `smd` file from your `stl` files: https://seissol.readthedocs.io/en/latest/simmodelerCAD-workflow.html
* Build PUMGen with simmetrix support: https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen#simmetrix-optional
* `pumgen -s simmodsuite -l /path/to/license/file mesh_file.smd`

**Reply by FURUNZ** [reply]
how to obtain a license

**Reply by Thomas-Ulrich** [reply]
info@simmetrix.com

**Comment by FURUNZ** [comment]
I found that I need a sign in the support download section, but I can't find where to sign up




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月29日(星期五) 下午5:01
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [SeisSol/SeisSol] How to mesh with SimModeler  using .stl file (Discussion #910)














 
***@***.***

**Reply by Thomas-Ulrich** [reply]
you need to ask simmtrix for an academic licence at [info@simmetrix.com](mailto:info@simmetrix.com)
Then they will send you some login information
